### PR TITLE
fix: switch to @node-ipc/compat

### PR DESCRIPTION
**Pull request by AaronDewes**
**What kind of change does this PR introduce?** (check at least one)

- [x] Bugfix
- [ ] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Docs
- [x] Underlying tools
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

**Other information:**

This pull request switches to [@node-ipc/compat](https://www.npmjs.com/package/@node-ipc/compat), a fork of node-ipc without malware.

### CODE REVIEW

**Comment by achrinza** [comment]
Looks like there's another `node-ipc` fork as well. :P

Just for reference, I'm also maintaining my own fork for v9 and v10/v11 here: https://github.com/achrinza/node-ipc/issues/1

File: packages/@vue/cli-shared-utils/lib/ipc.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-const ipc = require('node-ipc')
+const ipc = require('@node-ipc/compat')
```
Comment (Sacramentix) [inline packages/@vue/cli-shared-utils/lib/ipc.js:1]: Meaby add a little comment explaning why the node ipc has been replace

File: packages/@vue/cli-ui/apollo-server/util/ipc.js [hunk 2]
```diff
@@ -1,4 +1,4 @@
-const ipc = require('node-ipc')
+const ipc = require('@node-ipc/compat')
```
Comment (Sacramentix) [inline packages/@vue/cli-ui/apollo-server/util/ipc.js:1]: Meaby add a little comment here too explaning why the node ipc has been replace

**Review by Sacramentix — CHANGES_REQUESTED** [review]
Has a i said in comment, we could add a little comment explaining why the depedency has change.

**Comment by DumbGameMaker** [comment]
@node-ipc/compat seems like just a backported version of node-ipc (@node-ipc/node-ipc) not a fork. I could be wrong though. If this is true, might be a better option to choose a third-party fork like achrinza's, because the author of node-ipc could add the patch to @node-ipc/compat

**Comment by achrinza** [comment]
@DumbGameMaker They're all third-party forks of the original package `node-ipc` with different goals:

- **`@achrinza/node-ipc`:** Maintained by achrinza. Follows upstream closely. Drop-in replacement. No new features. 
    - **`@achrinza/node-ipc@9`:** Compatible fork of v9
    - **`@achrinza/node-ipc@10`:** Compatible fork of v10/v11
- **`@node-ipc/*`**: Maintained by AaronDewes. New features and enhancements such as TypeScript rewrite and use of Promises.
    - **`@node-ipc/compat@9`:** Compatible fork of v9
    - **`@node-ipc/node-ipc@11`:** Compatible fork of v10/v11

The [FAQ](https://github.com/achrinza/node-ipc/issues/1) provides more info on the difference.

**Comment by haoqunjiang** [comment]
Hi, thank you for all the help here!

After reading the FAQ section by @achrinza and reviewing the two projects, I decided to use `@achrinza/node-ipc` for now. https://github.com/vuejs/vue-cli/commit/75826d6598e7c17f727a891432d1902b9f31db0e

It's not that the other fork isn't good.
But the drop-in replacement approach is easier for me to review:
https://github.com/achrinza/node-ipc/compare/84dce1b4b28ac4fc66179ee3dcd5bc6d6c1032f1...v9
https://github.com/node-ipc/node-ipc/compare/97f4f49b63130033e67deeac424e691a8cc33f8c...d805d592f4e3b963126baea69f52ced569f6252f

And Vue CLI doesn't need any new feature from `node-ipc` for now, so I've temporarily pinned the dependency version, too.

Thanks again!

**Pull request closed without merging**